intl_translation_format/README.md

```diff
@@ -56,6 +56,169 @@ pub run intl_translation_format:generate
 ```
 
 ### Implement a new translation file format
-[TBD]
 
+The intl_translation_format brings the tools needed to support a new translation format. 
+You can extend the class TranslationFormat and complete the functions that will take care of the extracting of the messages
+and create templates. There are several classes to make this task easier: `MonoLingualFormat`, `BinaryMonolingualFormat`,
+`MultiLingualFormat`.
 
+As an example, this guide will saw you how the JSON file format is implemented.
+
+1. Learning about your format
+
+It is important to do some research about the format you are interested to implement. Some good questions would be:
+ 1. Is it a standard? 
+ 2. Is it a binary or text-based format?
+ 3. Does it contain a translation for multiple languages per file or is it just one language per file?
+ 4. How does it handle Plurals, Genders, and Variables?
+
+Let's answer these questions for our new JSON format:
+ 1. While it is a standard data interchange format, there is no official standard for translations. Multiple frameworks create their implementations like `i18next` or `ng2-translate` for `Angular`.
+
+ Because the `ICU Format` is the core of the dart intl library, we decided to create a value/key simple JSON with ICU messages. 
```

Comment:
Probably should have links. https://unicode-org.github.io/icu/userguide/format_parse/messages/ is probably good for the ICU format.  Although I just stumbled across https://phrase.com/blog/posts/guide-to-the-icu-message-format/ and skimmed it. Looks like it might be helpful, and even mentions Dart.